### DISCUSSION TITLE
Alldocube iPlay40 Pro Root 教程

### DISCUSSION THREAD

**Discussion by giscus** [kxxt/kxxt-website#128] [Thread]
# Alldocube iPlay40 Pro Root 教程

本文讲述了如何在酷比魔方 iPlay40 Pro 上安装 Magisk.

https://www.kxxt.dev/blog/how-to-root-iplay40pro/

**Comment by ephemeris26** [comment]
hey bro, I thank to you for this tutorial. finally I can root my Alldocube Iplay40 pro, but I faced a problem, after the device booting, it can read the battery. it just show 0% battery although I charge untill full. I hope you read and reply this comment. thx

**Reply by ephemeris26** [reply]
"it cannot"

**Reply by kxxt** [reply]
Hi @ephemeris26, I am glad to hear that you can finally root your device. As for the 0% battery bug, I am sorry that I can't offer much help here because I have never encountered such bugs. 

Maybe it has something to do with `/data/system/batterystats.bin` and you can find some solutions by googling "android battery calibration" and "android reset battery stats".

**Reply by ephemeris26** [reply]
I was find a solution, just make boot patch with magisk latest build.
And now I was successful flash gsi rom by following your tutorial bro, thx
I tried flash superior os and lineage os android 13, its can boot system but its some bug on camera. I also try flash pixel experience plus and nippon miui gsi rom, but I cann't resize the partition of that gsi file. It show 'unrecognised disk labe' .
But thx a alot for your hard work, I'm so waiting to find a way to flash twrp for T1020H x_x

**Comment by tbcteemo** [comment]
hello大佬，首先非常感谢你的教程。
我的平板的型号是smileX（T1028），按照你的教程进行到了最后给magisk-xxx的这个boot文件重新签名的步骤了。但是iplay和smileX型号不同，这个命令也没法通用，可以教一下在smileX上应该怎么改写这个命令吗？

**Comment by VanessaEvo** [comment]
Hi, first thanks for the tutorial. I've followed your tutorial, but I'm having trouble. I'm confused by the part "The private key file used to create your own private key, or the one located in the modified_fastboot directory used when unlocking the bootloader rsa4096_vbmeta.pem is referred to below as the private key file key.pem". When unlocking my device's bootloader, I used this repo https://github.com/TomKing062/CVE-2022-38694_unlock_bootloader and idk the key I used so I followed this tutorial https://www.hovatek.com/forum/thread-32662.html to create a private key and continue the tutorial you provided, but after I flashed these two files, magisk_patched and custom vbmeta, my device bootlooped. Did I do something wrong ?

**Comment by minhkhoi89** [comment]
Thanks for sharing, this method rooted my iplay 40 pro. First time experiencing with unisoc devices modding and spd flashing tools, it's nearly impossible to find any English guide on the internet for this device.
To sum it up:
1. [unlock the device boot loader](https://www.hovatek.com/forum/thread-32287.html) (the "unlocking bootloader..." can take up to 30 minutes waiting)
2. using this rooting method: generate vbmeta-custom-sign.img and magisk_patched_boot.img, then flash them to the device through spd-research-tool.
3. install gsi custom rom (anything arm64_bvN.img runs fine on my device, android 13+ has camera blurring issue), gsi custom rom can be installed after bootloader unlocked and before rooting, as I tested: 
adb reboot fastboot  (boot into device's fastbootd menu)
fastboot flash system lineageOS19.1_arm64_bvN.img   (fastboot delete logical partition system vendor _a/_b if needs more spaces here)
fastboot -w
After that, never do "fastboot reboot" to avoid boot loop, must select system reboot from the device's fastbootd menu.

And the stock rom .pac file is [here ](https://www.alldocube.com/en/software/14911/)

**Comment by zhushiSHR** [comment]
外网下载挂梯子太恼火了，给大家补个链接，有iplay40pro国外版系统，https://pan.shihaoran.cn/disk/s/ctnUSNE2rjM
